dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -1516,4 +1520,84 @@ public boolean canDataRead(IdentifiableObject identifiableObject) {
     return !aclService.isSupported(identifiableObject)
         || aclService.canDataRead(CurrentUserUtil.getCurrentUserDetails(), identifiableObject);
   }
+
+  @Override
+  @Transactional
+  public boolean sendEmailVerificationToken(User user, String requestUrl) {
+    String token = CodeGenerator.getRandomSecureToken();
+
+    String encodedToken = token + "|" + (System.currentTimeMillis() + EMAIL_TOKEN_EXPIRY_MILLIS);
+    user.setVerifiedEmail(encodedToken);
+    updateUser(user);
```

Comment:
Should we update `User` after email is sent successfully ?

Reply:
I don't think that adds much value. The different responses from email sending is not really 100% reliable anyway. It's worse that we end up NOT saving the token and maybe the email is sent, because the error messages is not correct. For example, it can be waiting in the transport and get sent later.
Worst case it is saved and not sent, is that nothing happens, and the user can try to send it again.